MonoGame.Framework.Content.Pipeline/Graphics/DxtTexture/BlockWindow.cs

```diff
@@ -0,0 +1,612 @@
+using System;
+
+using Vec3 = Microsoft.Xna.Framework.Vector3;
+
+namespace TextureSquish
+{
+    sealed class BlockWindow
+    {
+        public BlockWindow(Byte[] array, CompressionMode flags)
+        {
+            _Array = array;
+
+            flags = flags.FixFlags();

```

Comment:
You should store the flags here on the BlockWindow so a) you never have to pass it again in the Compress calls, b) you don't have to fix them on every block.